python/examples/running-ase-md.py

```diff
@@ -0,0 +1,254 @@
+# Molecular Dynamics (MD) simulation using ASE. ASE is not the only way to run MD with
+# metatensor models, but it is very easy to setup and use.
+#
+# .. TODO: once we have more MD engine, add link to their doc here
+
+from typing import Dict, List, Optional
+
+# tools to run the simulation
+import ase.build
+import ase.md
+import ase.md.velocitydistribution
+import ase.units
+
+# tools for visualization
+import ase.visualize.plot
+import matplotlib.pyplot as plt
+
+# the usuals suspects
+import numpy as np
+import torch
+
+from metatensor.torch import Labels, TensorBlock, TensorMap
+from metatensor.torch.atomistic import (
+    MetatensorAtomisticModel,
+    ModelCapabilities,
+    ModelOutput,
+    System,
+)
+
+# Integration with ASE calculator for metatensor atomistic models
+from metatensor.torch.atomistic.ase_calculator import MetatensorCalculator
+
+
+# %%
+#
+# The energy model
+# ----------------
+#
+# To run simulations, we'll need a model. For simplicity and keeping this tutorial
+# short, we'll use a model for `Einstein's solid`_, where all atoms oscillate around
+# their equilibrium position as harmonic oscillators, but don't interact with one
+# another. The energy of each atoms is given by:
+#
+# .. math::
+#
+#     E_i = k * (\vec{r_i} - \vec{r_i}^0)^2,
+#
+# where :math:`k` is a harmonic force constant, :math:`\vec{r_i}` the position of atom
+# :math:`i` and :math:`\vec{r_i}^0` the position of atom :math:`i` at equilibrium. The
+# energy of the whole system is then
+#
+# .. math::
+#
+#     E = \sum_i E_i.
+#
+# .. _Einstein's solid: https://en.wikipedia.org/wiki/Einstein_solid
+#
+# Let's implement this model! You can check the :ref:`atomistic-tutorial-export`
+# tutorial for more information on how to define your own custom models.
+
+
+class HarmonicModel(torch.nn.Module):
+    def __init__(self, k: float, equilibrium_positions: torch.Tensor):
+        super().__init__()
+        self.k = k
+        self.equilibrium_positions = equilibrium_positions
+
+    def forward(
+        self,
+        systems: List[System],
+        outputs: Dict[str, ModelOutput],
+        selected_atoms: Optional[Labels],
+    ) -> Dict[str, TensorMap]:
+        # if the model user did not request an energy calculation, we have nothing to do
+        if "energy" not in outputs:
+            return {}
+
+        # we don't want to worry about selected_atoms yet
+        if selected_atoms is not None:
+            raise NotImplementedError("selected_atoms is not implemented")
+
+        # compute the energy for each system by adding together the energy for each atom
+        energy = torch.zeros((len(systems), 1), dtype=systems[0].positions.dtype)
+        for i, system in enumerate(systems):
+            assert len(system) == self.equilibrium_positions.shape[0]
+            energy[i] += torch.sum(
+                self.k * (system.positions - self.equilibrium_positions) ** 2
+            )
```

Comment:
Do you actually check or use the `units` or `per_atom` property of `ModelOutput` here. Doesn't seem to be the case?

Reply:
you don't need to check the `units`, I've added something checking `per_atom` (even if it's going to be checked already by `MetatensorAtomisticModel`, this remind people they can handle this case).